Doc/c-api/arg.rst

```diff
@@ -191,7 +191,7 @@ which disallows mutable objects such as :class:`bytearray`.
    The buffer may contain embedded null bytes. The caller have to call
    :c:func:`PyBuffer_Release` when it is done with the buffer.
 
-``es`` (:class:`str`) [const char \*encoding, char \*\*buffer]
+``es`` (:class:`str`) [*encoding*, char \*buffer]
```

Comment:
Why do you remove "const char*" type on encoding: the type is correct, no?

Reply:
I suspect that this is following [Martin's comment](https://github.com/python/cpython/issues/68168#issuecomment-1093681003) on the issue.

Reply:
Ok, but Martin's comment also says:

> The text description should explain what *encoding* is, but it appears it may already do that well enough.

Reply:
@erlend-aasland: Do you want to address this issue?

Reply:
Sure, I can take care of that later today.

Reply:
I believe the arguments are explained pretty good already:

>    This format requires two arguments.  The first is only used as input, and 
   must be a :c:expr:`const char*` which points to the name of an encoding as a
   NUL-terminated string, or ``NULL``, in which case ``'utf-8'`` encoding is used.
   An exception is raised if the named encoding is not known to Python.  The 
   second argument must be a :c:expr:`char**`; the value of the pointer it
   references will be set to a buffer with the contents of the argument text.
   The text will be encoded in the encoding specified by the first argument.

Reply:
But, I find it strange that the type is removed. Passing a `NULL` argument to a `const char *` parameter is perfectly allowed in C.

Reply:
Reading again the OP by Larry, I'm not so sure he meant the data types to be removed: https://github.com/python/cpython/issues/68168#issuecomment-1093680996

Reply:
See also the description of the table at the top of `arg.rst`:

> In the following description, the quoted form is the format
unit; the entry in (round) parentheses is the Python object type that matches
the format unit; and the entry in [square] brackets is the type of the C
variable(s) whose address should be passed.